File: src/modifiers/buttonSizing.ts
import { createModifier } from './createModifier';

/// How a button sizes itself within its container.
/// - `'fitted'` hugs the title
/// - `'flexible'` expands to fill the available width
export type ButtonSizingValue = 'automatic' | 'fitted' | 'flexible';

/// SwiftUI `.buttonSizing(_:)`.
///
/// LIMITATION: `.buttonSizing(_:)` only exists on watchOS 26+ SDKs, which is
/// newer than this package's build floor. To keep the native side compiling on
/// every supported SDK, the modifier is currently a documented no-op — the view
/// is returned unchanged. The factory still ships so callers can adopt it
/// without churn once the native side raises its SDK and wires it up.
export function buttonSizing(sizing: ButtonSizingValue = 'automatic') {
  return createModifier('buttonSizing', { sizing });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/ButtonSizing.swift
import SwiftUI

/// SwiftUI `.buttonSizing(_:)` — controls whether a button hugs its title
/// (`.fitted`) or expands to fill available width (`.flexible`).
///
/// LIMITATION: `.buttonSizing(_:)` (and the `ButtonSizing` type) is only
/// available on watchOS 26+ SDKs, newer than this package's build floor
/// (watchOS 9). Referencing the symbol — even behind `#available` — would fail
/// to compile on the supported build SDKs, so this is a documented no-op: the
/// view is returned unchanged. There is no stable equivalent on watchOS 9.
/// Registering the `$type` keeps dispatch consistent and lets the no-op be
/// swapped for a real implementation once the SDK floor is raised.
enum RNWButtonSizingModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("buttonSizing") { view, _, _ in view }
    }
}
